superset/models/slice.py

```diff
@@ -321,3 +343,8 @@ def set_related_perm(mapper, connection, target):
     sqla.event.listen(Slice, "after_insert", ChartUpdater.after_insert)
     sqla.event.listen(Slice, "after_update", ChartUpdater.after_update)
     sqla.event.listen(Slice, "after_delete", ChartUpdater.after_delete)
+
+# events for updating tags
+if is_feature_enabled("THUMBNAILS_SQLA_LISTENERS"):
```

Comment:
Is there a way to drag this out of the model? I see this as debt that will need to be paid down in the future, as models should themselves be as lean as possible.

Reply:
Agree, yes. Promise to clean this debt when refactoring Chart's API to comply with SIP-35

Reply:
😆